### ISSUE TITLE
Don't append extension if the current word already has one

### ISSUE THREAD

**Issue by chefarov** [OJ/gobuster#144]
Some wordlists contain filenames (e.g `status.php`) along with directories. So if someone runs 

    gobuster -x php -w mixed-wordlist.txt  ...

then a redundant request to `/status.php.php` will be made. Nothing bad happens, however we increase a bit the saturation in the network links  :)

Do you think adding a condition to prevent it, would be a good idea?

Congrats for the program, its great.

**Comment by OJ** [comment]
Hi @chefarov,

Personally, no I do not think we should try to get smart with file names. It's actually really hard to get right, believe it or not there have been a fair few times when requests in the form `foo.txt.txt` have resulted in a win for me.

The question of "how smart should we get with wordlists" is a tough one to answer in a way that keeps everyone happy. Ultimately the best way is to remain completely dumb. It keeps the code clean and simple, it keeps the application going as fast as it can, and it completely avoids cases where users get confused as to why certain things are stripped and other things aren't. Everyone understands how it works, and why. I think the responsibility should lie on the user to provide the right wordlist with whatever filtering already applied. Don't want `.php` on the end? Then remove it before you throw it at the tool.

So, in short, no I am not really open to munging wordlists as we parse them.

Does that make sense? Cheers mate!

**Comment by chefarov** [comment]
Thank you for your comment

Yeah makes sense :) I suppose its cleaner to pre-process the wordlist each time according to the target.
